tests/v2/validation/fleet/provisioning/new_cluster_existing_gitrepo_test.go

```diff
@@ -0,0 +1,301 @@
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, userConfig)
+
+	f.provisioningConfig = userConfig
+
+	if f.provisioningConfig.RKE2KubernetesVersions == nil {
+		rke2Versions, err := kubernetesversions.ListRKE2AllVersions(f.client)
+		require.NoError(f.T(), err)
+
+		f.provisioningConfig.RKE2KubernetesVersions = []string{rke2Versions[len(rke2Versions)-1]}
+	}
+
+	if f.provisioningConfig.CNIs == nil {
+		f.provisioningConfig.CNIs = []string{cniCalico}
+	}
+
+	enabled := true
+	var testuser = namegen.AppendRandomString("testuser-")
+	var testpassword = password.GenerateUserPassword("testpass-")
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(f.T(), err)
+
+	newUser.Password = user.Password
+
+	standardUserClient, err := client.AsUser(newUser)
+	require.NoError(f.T(), err)
+
+	f.standardUserClient = standardUserClient
+}
+
+func (f *FleetWithProvisioningTestSuite) TestProvisionAfterAddedGitRepo() {
+	fleetVersion, err := fleet.GetDeploymentVersion(f.client, fleet.FleetControllerName, localName)
+	require.NoError(f.T(), err)
+
+	chartVersion, err := f.client.Catalog.GetLatestChartVersion(fleetName, catalog.RancherChartRepo)
+	require.NoError(f.T(), err)
+
+	// fleet chart version may contain chart version info that is a superset of the version reported by the fleet deployment.
+	require.Contains(f.T(), chartVersion, fleetVersion[1:])
+
+	nodeRolesDedicated := []provisioninginput.MachinePools{
+		provisioninginput.EtcdMachinePool,
+		provisioninginput.ControlPlaneMachinePool,
+		provisioninginput.WorkerMachinePool,
+	}
+
+	tests := []struct {
+		name         string
+		client       *rancher.Client
+		machinePools []provisioninginput.MachinePools
+		isWindows    bool
+		runFlag      bool
+	}{
+		{fleetName + " " + fleetVersion, f.standardUserClient, nodeRolesDedicated, false, f.client.Flags.GetValue(environmentflag.Long)},
+	}
+
+	for _, tt := range tests {
+		if !tt.runFlag {
+			f.T().Logf("SKIPPED")
+			continue
+		}
+
+		testSession := session.NewSession()
+		defer testSession.Cleanup()
+
+		tt.client, err = tt.client.WithSession(testSession)
+		require.NoError(f.T(), err)
+
+		adminClient, err := f.client.WithSession(testSession)
+		require.NoError(f.T(), err)
+
+		provisioningConfig := *f.provisioningConfig
+		provisioningConfig.MachinePools = tt.machinePools
+
+		f.Run(tt.name, func() {
+			logrus.Info("Deploying public fleet gitRepo")
+			gitRepoObject, err := extensionsfleet.CreateFleetGitRepo(adminClient, f.fleetGitRepo)
+			require.NoError(f.T(), err)
+
+			logrus.Info("Deploying Custom Cluster")
+			testClusterConfig := clusters.ConvertConfigToClusterConfig(&provisioningConfig)
+
+			_, _, customProvider, _ := permutations.GetClusterProvider(permutations.RKE2CustomCluster, f.provisioningConfig.NodeProviders[0], &provisioningConfig)
```

Comment:
I noticed that the `GetClusterProvider` function is the only one being used from the permutations file in the Fleet provisioning tests. Instead of moving the entire permutations package to actions, what do you think about moving just the GetClusterProvider function to actions?

do we anticipate any current or future tests that might require other functions from the permutations file? This could help decide if we want to keep the package together or only move what's necessary.

Reply:
I think it makes the most sense to move the whole package to actions but this is mostly why I added Sam + had conversations with Israel earlier this week about the design. I don't think splitting the package makes sense to do.

Reply:
I understand the idea of keeping the whole permutations package together, but considering that actions are intended for common operations used by tests, I’m concerned that we might end up making the actions package bulky by moving whoel of permuations. So far, we’re only using GetClusterProvider in the Fleet provisioning tests, which makes me lean toward moving just this function to actions while leaving the rest of permutations.

Would splitting the package here add more complexity, or is it more a matter of keeping everything consistent for now? I just want to make sure, we’re not overloading the actions package with functions that don’t necessarily fit the intend purpose.

Reply:
I think actions is meant to be 'bulky' compared to validations. Moving forward, anything shared should go in actions instead of hidden in a test package.
splitting permutations would add complexity and is more consistent to keep it all together in one package.